### PR TITLE
Refactor engine and add leaderboard

### PR DESCRIPTION
**Pull request by Andrei-Dolgolev**

### CODE REVIEW

File: dev.sh [hunk 1]
```diff
@@ -6,4 +6,4 @@ set -e
 
 LOOTBOX_PORT="${LOOTBOX_PORT:-7191}"
```
Comment (kompotkot) [inline dev.sh:8]: - In deploy/engine.service lootbos should be changed to engineapi
https://github.com/bugout-dev/engine/blob/991c9a545e8c64239b1dd7c1bf34552f322d311a/deploy/engine.service#L14

- Everything renamed in setup.py
https://github.com/bugout-dev/engine/blob/991c9a545e8c64239b1dd7c1bf34552f322d311a/setup.py#L11

**Review by kompotkot — CHANGES_REQUESTED** [review]

File: dev.sh (hunk 1 above)
Comment (kompotkot) [reply dev.sh:8]: And if you started changing dev.sh, so  pls change to the end LOOTBOX_PORT, etc :)

**Review by kompotkot — APPROVED** [review]
lg

**Review by zomglings — APPROVED** [review]
gg

Just ran migrations on prod database.

**Pull request merged**